internal/api/front/perses/src/app/project/dashboard/dashboard.guard.ts

```diff
@@ -0,0 +1,59 @@
+// Copyright 2021 Amadeus s.a.s
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, CanActivate, Router, RouterStateSnapshot } from '@angular/router';
+import { Observable, of } from 'rxjs';
+import { ProjectService } from '../project.service';
+import { catchError, concatMap, map, tap } from 'rxjs/operators';
+import { ToastService } from '../../shared/service/toast.service';
+import { CustomError } from '../../shared/model/error.model';
+import { DashboardService } from './service/dashboard.service';
+import { CanActivateReturnType } from '../../shared/utils/types.utils';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class DashboardGuard implements CanActivate {
+  constructor(private readonly dashboardService: DashboardService,
+              private readonly projectService: ProjectService,
+              private readonly toastService: ToastService,
+              private readonly router: Router) {
+  }
+
+  /**
+   * This guard is used to trigger the load of the list of available dashboards
+   * for the current project into the cache as soon as the dashboard module is
+   * reached.
+   */
+  canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): CanActivateReturnType {
+    let currentProject: string | undefined;
+    return (this.projectService.getCurrent() || of()).pipe(
```

Comment:
I think you should also include in this guard when the user is getting a single dashboard. So the guard can redirect the user to the project page if the dashboard doesn't exist

Reply:
sry I dont understand what you mean here :D

Reply:
yeah my comment is not well placed.

So here your gard is getting the list of the dashboard which is cool :). But when you are going to ``/projects/perses/dashboards/SimpleDashboard` for example, your guard could also check that the `SimpleDashboard` exists so you won't be able to access this page if `SimpleDashboard` doesn't exist.

Is it clearer saying like that ?

Reply:
aaah yes you're right I'm not checking that case, I will add this

Reply:
Actually it maybe more interesting to create another guard that would only be applied when we are going to check the dashboard-details.

Like that you won't have to touch this guard and you can basically chain them.
WDYT ?

Reply:
Added : https://github.com/perses/perses/pull/41/commits/664a657c71e2572951dae77739da2ea6442ef5cc